pkg/didmethod/peer/did.go

```diff
@@ -30,7 +31,60 @@ func newDid(doc *did.Doc) (string, error) {
 	if doc.PublicKey == nil || doc.Authentication == nil {
 		return "", errors.New("the genesis version must include public keys and authentication")
 	}
+	numBasis, err := numBasis(doc)
+	if err != nil {
+		return "", err
+	}
+	messageIdentifier := []string{peerPrefix, numAlgo, encAlgo, "-", numBasis}
+	peerDID := strings.Join(messageIdentifier, "")
 
+	return peerDID, nil
+}
+
+/*
+ validateDid checks the format of the doc's DID and checks that the DID's 'namestring' matches against its numeric basis as per the
+ Namestring Generation Method: https://openssi.github.io/peer-did-method-spec/index.html#namestring-generation-method
+ Note: this check should be done only on the resolved variant of the genesis version of Peer DID documents.
+*/
+func validateDid(doc *did.Doc) error {
```

Comment:
DID should be capitals: validateDID

https://github.com/golang/go/wiki/CodeReviewComments#initialisms